File: convert_videos.py
#!/usr/bin/env python3
"""
Converte todos os arquivos .MOV em Galeria/ para .mp4
Requer ffmpeg instalado no PATH do sistema ou moviepy
"""
import os
import subprocess
import sys
from pathlib import Path

def convert_mov_to_mp4_ffmpeg(input_path, output_path):
    """Converte MOV para MP4 usando ffmpeg"""
    try:
        cmd = [
            'ffmpeg',
            '-i', str(input_path),
            '-c:v', 'libx264',       # H.264 codec
            '-preset', 'fast',       # velocidade
            '-crf', '23',            # qualidade (0-51, 23 é default)
            '-c:a', 'aac',           # áudio AAC
            '-b:a', '128k',          # bitrate áudio
            '-y',                    # overwrite sem perguntar
            str(output_path)
        ]
        print(f"Convertendo {input_path.name} -> {output_path.name}...")
        result = subprocess.run(cmd, capture_output=True, text=True, timeout=300)
        if result.returncode == 0:
            print(f"  ✓ Sucesso: {output_path.name}")
            return True
        else:
            print(f"  ✗ Erro: {result.stderr}")
            return False
    except FileNotFoundError:
        print("ffmpeg não encontrado. Tentando moviepy...")
        return False
    except Exception as e:
        print(f"  ✗ Erro ao converter: {e}")
        return False

def convert_mov_to_mp4_moviepy(input_path, output_path):
    """Alternativa: converte MOV para MP4 usando moviepy"""
    try:
        from moviepy.editor import VideoFileClip
        print(f"Convertendo (moviepy) {input_path.name} -> {output_path.name}...")
        clip = VideoFileClip(str(input_path))
        clip.write_videofile(str(output_path), verbose=False, logger=None)
        clip.close()
        print(f"  ✓ Sucesso: {output_path.name}")
        return True
    except ImportError:
        print("moviepy não instalado. Execute: pip install moviepy")
        return False
    except Exception as e:
        print(f"  ✗ Erro ao converter: {e}")
        return False

def main():
    galeria_dir = Path("Galeria")
    if not galeria_dir.exists():
        print("Erro: Diretório 'Galeria' não encontrado")
        return False
    
    mov_files = list(galeria_dir.glob("*.MOV")) + list(galeria_dir.glob("*.mov"))
    if not mov_files:
        print("Nenhum arquivo .MOV encontrado em Galeria/")
        return True
    
    print(f"Encontrados {len(mov_files)} arquivo(s) MOV\n")
    
    success_count = 0
    failed_count = 0
    
    for mov_file in mov_files:
        mp4_file = mov_file.with_suffix(".mp4")
        
        # Tenta ffmpeg primeiro
        if convert_mov_to_mp4_ffmpeg(mov_file, mp4_file):
            success_count += 1
        else:
            # Tenta moviepy como alternativa
            if convert_mov_to_mp4_moviepy(mov_file, mp4_file):
                success_count += 1
            else:
                failed_count += 1
    
    print(f"\n=== Resumo ===")
    print(f"Convertidos com sucesso: {success_count}")
    print(f"Falharam: {failed_count}")
    
    return failed_count == 0

if __name__ == "__main__":
    success = main()
    sys.exit(0 if success else 1)

File: convert_videos.js
#!/usr/bin/env node
/**
 * Converte arquivos .MOV para .mp4 usando ffmpeg-static
 * Uso: node convert_videos.js
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const GALERIA_DIR = path.join(__dirname, 'Galeria');

function log(msg) {
  console.log(`[${new Date().toLocaleTimeString()}] ${msg}`);
}

function ensureFFmpeg() {
  try {
    execSync('ffmpeg -version', { stdio: 'pipe' });
    log('✓ ffmpeg encontrado no PATH');
    return 'ffmpeg';
  } catch (e) {
    log('✗ ffmpeg não encontrado, tentando instalar ffmpeg-static...');
    try {
      execSync('npm install ffmpeg-static --save-dev', { 
        cwd: __dirname,
        stdio: 'pipe'
      });
      const ffmpegPath = require('ffmpeg-static');
      log(`✓ ffmpeg-static instalado: ${ffmpegPath}`);
      return ffmpegPath;
    } catch (err) {
      log('✗ Falha ao instalar ffmpeg-static: ' + err.message);
      throw err;
    }
  }
}

function convertMOVtoMP4(ffmpegPath, inputFile, outputFile) {
  try {
    const cmd = `"${ffmpegPath}" -i "${inputFile}" -c:v libx264 -preset fast -crf 23 -c:a aac -b:a 128k -y "${outputFile}"`;
    
    log(`Convertendo: ${path.basename(inputFile)} -> ${path.basename(outputFile)}`);
    execSync(cmd, { stdio: 'pipe' });
    
    // Verifica se o arquivo foi criado
    if (fs.existsSync(outputFile)) {
      const inputSize = fs.statSync(inputFile).size;
      const outputSize = fs.statSync(outputFile).size;
      const ratio = ((outputSize / inputSize) * 100).toFixed(1);
      log(`  ✓ Sucesso! Tamanho: ${(inputSize/1024/1024).toFixed(1)}MB → ${(outputSize/1024/1024).toFixed(1)}MB (${ratio}%)`);
      return true;
    } else {
      log(`  ✗ Arquivo de saída não foi criado`);
      return false;
    }
  } catch (err) {
    log(`  ✗ Erro: ${err.message.split('\n')[0]}`);
    return false;
  }
}

async function main() {
  try {
    log('Iniciando conversão de vídeos MOV para MP4...\n');

    // Garante que ffmpeg está disponível
    const ffmpegPath = ensureFFmpeg();
    log('');

    // Lista arquivos MOV
    if (!fs.existsSync(GALERIA_DIR)) {
      log('✗ Diretório Galeria não encontrado');
      process.exit(1);
    }

    const files = fs.readdirSync(GALERIA_DIR);
    const movFiles = files.filter(f => f.toLowerCase().endsWith('.mov'));

    if (movFiles.length === 0) {
      log('Nenhum arquivo .MOV encontrado');
      process.exit(0);
    }

    log(`Encontrados ${movFiles.length} arquivo(s) .MOV\n`);

    let successCount = 0;
    let failCount = 0;

    for (const movFile of movFiles) {
      const inputPath = path.join(GALERIA_DIR, movFile);
      const baseName = path.basename(movFile, path.extname(movFile));
      const outputPath = path.join(GALERIA_DIR, `${baseName}.mp4`);

      // Pula se o MP4 já existe
      if (fs.existsSync(outputPath)) {
        log(`⊘ Pulando (MP4 já existe): ${movFile}`);
        continue;
      }

      if (convertMOVtoMP4(ffmpegPath, inputPath, outputPath)) {
        successCount++;
      } else {
        failCount++;
      }
    }

    log(`\n=== Resumo ===`);
    log(`Convertidos com sucesso: ${successCount}`);
    log(`Falharam: ${failCount}`);
    
    process.exit(failCount > 0 ? 1 : 0);
  } catch (err) {
    log(`Erro fatal: ${err.message}`);
    process.exit(1);
  }
}

main();
